nxt/sensor/hitechnic.py

```diff
@@ -459,9 +460,238 @@ def set_digital_modes(self, modes):
         """Sets input/output mode of digital pins. Can take a Digital_Data() object."""
         self.write_value('digital_cont', (int(modes), ))
 
+
 Prototype.add_compatible_sensor(None, 'HiTechnc', 'Proto   ')
 
 
+class SuperPro(BaseDigitalSensor):
+    """
+    Object for HiTechnic sensor SuperPro boards.
+    """
+    I2C_ADDRESS = BaseDigitalSensor.I2C_ADDRESS.copy()
+    I2C_ADDRESS.update({
+        "version": (0x00, "8s"),
+        "manufacturer": (0x08, "8s"),
+        "sensor_type": (0x10, "8s"),
+        "analog_a0": (0x42, "<H"),
+        "analog_a1": (0x44, "<H"),
+        "analog_a2": (0x46, "<H"),
+        "analog_a3": (0x48, "<H"),
+        "digital_in": (0x4C, "<B"),
+        "digital_out": (0x4D, "<B"),
+        "digital_dir": (0x4E, "<B"),
+        "strobe_out": (0x50, "B"),
+        "led_out": (0x51, "B"),
+        "analog_out0_mode": (0x52, "B"),
+        "analog_out0_freq": (0x53, "<H"),
+        "analog_out0_volts": (0x55, "H"),
+        "analog_out1_mode": (0x57, "B"),
+        "analog_out1_freq": (0x58, "<H"),
+        "analog_out1_volts": (0x5A, "H")
+    })
+
+    """
+    Some users online report that this I2C device address is 0x08, the documentation states that it's 0x10.
+    Let us know if this value doesn't work for you and 0x08 does. 0x10 works in testing with my hardware.
+    """
+    I2C_DEV = 0x10
+
+    # Used for converting ADC bits into percent signal
+    ANALOG_LSB = 1 / 1023
+
+    class AnalogOutputMode(IntEnum):
+        DC = 0
+        SINE = 1
+        SQUARE = 2
+        UPWARDS_SAWTOOTH = 3
+        DOWNWARDS_SAWTOOTH = 4
+        TRIANGLE = 5
+        # There's supposed to be a PWM type, but it doesn't seem to function.
+
+    def get_analog(self) -> dict[str, int]:
+        """
+        Get analog input pins (A0-A3) result in raw bits
+        :return: Map of pin name, pin value (bits)
+        """
+        analog_in_pin_name_map = [("analog_a0", "a0"), ("analog_a1", "a1"), ("analog_a2", "a2"), ("analog_a3", "a3")]
+        analog_raw_map = {}
+        for pin in analog_in_pin_name_map:
+            raw = self.read_value(pin[0])[0]
+            # Not 100% sure about this logic but seems to work.
+            low = (raw & 0xFF00) >> 8
+            high = raw & 0x00FF
+            analog_input = low + high * 4
+            """
+            Debug printouts, in case you need them
+            print("raw {0:#016b}".format(raw))
+            print("low {0:#016b}".format(low))
+            print("high {0:#016b}".format(high))
+            print("{0}: {1} {1:#016b}".format(pin, analog_input))
```

Comment:
If you want to keep them, use logging (see nxt/locator.py for an example), else, drop it.

Reply:
Removed.